components/ui/useToast.tsx

```diff
@@ -0,0 +1,189 @@
+// Inspired by react-hot-toast library
+import * as React from 'react';
+
+import type { ToastActionElement, ToastProps } from './Toast';
+
+const TOAST_LIMIT = 3;
+const TOAST_REMOVE_DELAY = 1000000;
```

Comment:
It would be clearer to specify the unit in the variable name or as a comment

```suggestion
const TOAST_REMOVE_DELAY_IN_MS = 1000000;
```